File: app/src/main/java/ru/telematica/casco2go/model/eventbus/ErrorEvent.java
package ru.telematica.casco2go.model.eventbus;

import io.reactivex.annotations.Nullable;

/**
 * Created by m.sidorov on 29.04.2018.
 */

public class ErrorEvent {

    public final String message;
    public final @Nullable Throwable error;

    public ErrorEvent(String message, Throwable error){
        this.message = message;
        this.error = error;
    }

    public ErrorEvent(String message){
        this.message = message;
        this.error = null;
    }

}

File: app/src/main/java/ru/telematica/casco2go/model/eventbus/ErrorEvent.kt
package ru.telematica.casco2go.model.eventbus

import ru.telematica.casco2go.utils.isNull

/**
 * Created by m.sidorov on 29.04.2018.
 */
class ErrorEvent(val message: String, val error: Throwable?) {

    constructor(message: String) : this(message, error = null)
    constructor(error: Throwable) : this(message = "", error = error)

}